tools/slo-generator/slo_generator/backends/prometheus.py

```diff
@@ -70,42 +64,38 @@ def query_sli(self, **kwargs):
         LOGGER.debug(f"SLI value: {sli_value}")
         return sli_value
 
-    def good_bad_ratio(self, **kwargs):
+    def good_bad_ratio(self, timestamp, window, slo_config):
         """Compute good bad ratio from two metric filters.
 
         Args:
-            kwargs (dict):
-                window (str): Query window.
-                measurement (dict): Measurement config
-                    filter_good (str): PromQL query for good events.
-                    filter_bad (str, optional): PromQL query for bad events.
-                    filter_valid (str, optional): PromQL query for valid events.
+            timestamp (int): UNIX timestamp.
+            window (int): Window (in seconds).
+            slo_config (dict): SLO configuration.
 
         Note:
             At least one of `filter_bad` or `filter_valid` is required.
 
         Returns:
             tuple: A tuple of (good_event_count, bad_event_count).
         """
-        window = kwargs['window']
-        filter_good = kwargs['measurement']['filter_good']
-        filter_bad = kwargs['measurement'].get('filter_bad')
-        filter_valid = kwargs['measurement'].get('filter_valid')
+        conf = slo_config['backend']
+        filter_good = conf['measurement']['filter_good']
+        filter_bad = conf['measurement'].get('filter_bad')
+        filter_valid = conf['measurement'].get('filter_valid')
 
         # Replace window by its value in the error budget policy step
         expr_good = filter_good.replace('[window]', f'[{window}s]')
         res_good = self.query(expr_good)
-        good_event_count = PrometheusBackend.count(res_good)
+        good_event_count = PROM.count(res_good)
```

Comment:
I did this to shorten the line lengths which could be > 80 so that `pylint` tests would pass. 

Since there is no way to split the lines too long with '\' when calling a class method, we have two solutions:
- Ignore the line length errors in `pylint` and `flake`
- Shorten the class names (drop the `Backend` suffix ?)

Reply:
Fixed.